sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierTest.cs

```diff
@@ -32,6 +32,18 @@ public void RawIdTakesPrecendenceInEqualityCheck()
         public void MicrosoftTeamsUserIdentifier_DefaultCloudIsPublic()
             => Assert.AreEqual(CommunicationCloudEnvironment.Public, new MicrosoftTeamsUserIdentifier("45ab2481-1c1c-4005-be24-0ffb879b1130", isAnonymous: true, rawId: "Raw Id").Cloud);
 
+        [Test]
+        public void RawIdDeserializesToCorrectAcsCloudType()
+        {
+            Assert.AreEqual(CommunicationCloudEnvironment.Public, (CommunicationIdentifier.FromRawId("8:acs:bbbcbc1e-9f06-482a-b5d8-20e3f26ef0cd_45ab2481-1c1c-4005-be24-0ffb879b1130") as CommunicationUserIdentifier)?.Cloud);
```

Comment:
I wonder why the `?`. Doesn't the `FromRawId` method always return a value?

Reply:
It does. But it returns instance of `CommunicationIdentifier` type that doesn't contain `Cloud` property, so I need to cast it to `CommunicationUserIdentifier` that does. I guess it's then not able to infer that the property can't be `null`. Anyway, without the check I am getting a compiler error, so this was the way how to get rid of it :)